lib/library.rb

```diff
@@ -0,0 +1,48 @@
+class Library
+  attr_reader :name, :books, :authors, :checked_out_books
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []    
+    @checked_out_books = []
+    @checkout_log = []
+  end
+
+  def add_author(author)
+    #an add_book method is probably more proper as a library is unlikely to add all books of an author at once.
+    author.books.each { |book| @books << book }
+    @authors << author
+  end
+
+  def publication_time_frame_for(author)
+    time_frame = {}
+    time_frame[:start] = author.books.map {|book| book.publication_year}.min
+    time_frame[:end] = author.books.map {|book| book.publication_year}.max
+    time_frame
```

Comment:
Great job with this, however you have repeated code. You're mapping over author.books twice. Can dry that up by storing it in a variable and then calling .min and .max on that variable.

Reply:
omg, was not thinking of scalability here. Thank you for the advice!